server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java

```diff
@@ -191,6 +191,8 @@ public boolean transferSnapshot(TransferSnapshot transferSnapshot, TranslogTrans
             captureStatsOnUploadFailure();
             translogTransferListener.onUploadFailed(transferSnapshot, ex);
             return false;
+        } finally {
+            translogTransferListener.close();
```

Comment:
can we use `try(translogTransferListener)` so that it gets closed automatically ?

Reply:
Done